corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -398,19 +455,61 @@ def create_report(self):
         report = ReportConfiguration(
             domain=self.domain,
             config_id=data_source_config_id,
-            title=self.cleaned_data['report_name'],
+            title=self.report_name,
             aggregation_columns=self._report_aggregation_cols,
             columns=self._report_columns,
             filters=self._report_filters,
-            configured_charts=self._report_charts
+            configured_charts=self._report_charts,
+            report_meta=ReportMeta(
+                created_by_builder=True,
+                builder_report_type=self.report_type
+            )
         )
         report.validate()
         report.save()
         return report
 
     @property
+    @memoized
+    def initial_filters(self):
+        if self.existing_report:
+            return [self._get_view_model(f) for f in self.existing_report.filters]
+        if self.source_type == 'case':
+            return [
+                {'property': 'closed', 'display_text': 'closed', 'format': 'Choice'},
```

Comment:
+1 if it's part of a larger schema. If you just want to have this be a sort of "result object" that you pass around, I'd just use a `namedtuple` to define that schema.